bundles/AdminBundle/Controller/Searchadmin/SearchController.php

```diff
@@ -478,14 +481,25 @@ public function quicksearchAction(Request $request, EventDispatcherInterface $ev
                     'fullpath' => htmlspecialchars($element->getRealFullPath()),
                     'fullpathList' => htmlspecialchars($this->shortenPath($element->getRealFullPath())),
                     'iconCls' => 'pimcore_icon_asset_default',
+                    'icon' => '',
                 ];
 
                 if ($element instanceof Asset) {
                     $data['iconCls'] .= ' pimcore_icon_' . \Pimcore\File::getFileExtension($element->getFilename());
+                 } elseif ($element instanceof DataObject\Concrete) {
+                    // Show custom on base class (not variants) if it exists
+                    if ($element->getType() === DataObject::OBJECT_TYPE_OBJECT && $element->getClass()->getIcon()) {
+                        $data['icon'] = $element->getClass()->getIcon();
+                        $data['iconCls'] = '';
+                    } else {
+                        $data['iconCls'] .= ' pimcore_icon_' . $element->getType();
+                    }
                 } else {
                     $data['iconCls'] .= ' pimcore_icon_' . $element->getType();
                 }
 
+                $this->addAdminStyle($element, null, $data);
```

Comment:
I think we should introduce a new context for the search `\Pimcore\Event\Admin\ElementAdminStyleEvent::CONTEXT_SEARCH` instead of passing `null`.